src/renderers/shared/fiber/ReactDebugCurrentFiber.js

```diff
@@ -34,7 +34,11 @@ function getCurrentFiberOwnerName() : string | null {
     switch (fiber.tag) {
       case IndeterminateComponent:
       case ClassComponent:
+        return getComponentName(fiber);
       case FunctionalComponent:
+        if (fiber._debugOwner != null) {
+          return getComponentName(fiber._debugOwner);
+        }
```

Comment:
This doesn't look right to me. There's nothing special about functional components from the warning message perspective as far as I know.